src/SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -216,30 +216,21 @@ private void PostProcessProjectStatus(IEnumerable<ProjectData> projects)
             }
         }
 
-        private static ProjectData GetClosestProjectOrDefault(string filePath, IEnumerable<ProjectData> projects)
+        private static ProjectData GetClosestProjectOrDefault(FileInfo fileInfo, IEnumerable<ProjectData> projects)
         {
             var longestMatchingPath = (Length: 0, Items: new List<ProjectData>());
 
             foreach (var project in projects)
             {
-                var projectPath = project.Project.GetProjectDirectory();
-
-                // Ensure that folder path always ends with backslash so a directory like 'c:\aaa\bbb' doesn't match
-                // a file like 'c:\aaa\bbb.cs' nor 'c:\aaa\bbbxxxx\myfile.cs'
-                if (!projectPath.EndsWith(Path.DirectorySeparatorChar.ToString()))
-                {
-                    projectPath += Path.DirectorySeparatorChar;
-                }
-
-                if (filePath.StartsWith(projectPath))
+                if (PathHelper.IsInDirectory(fileInfo, project.Project.GetDirectory()))
                 {
-                    if (filePath.Length == longestMatchingPath.Length)
+                    if (fileInfo.Length == longestMatchingPath.Length)
```

Comment:
This doesn't look right: _[FileInfo.Length](https://msdn.microsoft.com/en-us/library/system.io.fileinfo.length(v=vs.110).aspx)_ returns the length of the file, not the length of the file name. Same in line 231.
If the unit tests are all passing then there's probably some missing test coverage.

Reply:
Crap! Thanks for spotting this...